internal/command/jsonformat/state_test.go

```diff
@@ -0,0 +1,437 @@
+package jsonformat
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/mitchellh/colorstring"
+
+	"github.com/hashicorp/terraform/internal/command/format"
+	"github.com/hashicorp/terraform/internal/command/jsonprovider"
+	"github.com/hashicorp/terraform/internal/command/jsonstate"
+	"github.com/hashicorp/terraform/internal/states/statefile"
+	"github.com/hashicorp/terraform/internal/terminal"
+
+	"github.com/zclconf/go-cty/cty"
+
+	"github.com/hashicorp/terraform/internal/addrs"
+	"github.com/hashicorp/terraform/internal/configs/configschema"
+	"github.com/hashicorp/terraform/internal/providers"
+	"github.com/hashicorp/terraform/internal/states"
+	"github.com/hashicorp/terraform/internal/terraform"
+)
+
+func TestState(t *testing.T) {
+	color := &colorstring.Colorize{Colors: colorstring.DefaultColors, Disable: true}
+
+	tests := []struct {
+		State *format.StateOpts
+		Want  string
+	}{
+		{
+			&format.StateOpts{
+				State:   &states.State{},
+				Color:   color,
+				Schemas: &terraform.Schemas{},
+			},
+			"The state file is empty. No resources are represented.\n",
+		},
+		{
+			&format.StateOpts{
+				State:   basicState(t),
+				Color:   color,
+				Schemas: testSchemas(),
+			},
+			basicStateOutput,
+		},
+		{
+			&format.StateOpts{
+				State:   nestedState(t),
+				Color:   color,
+				Schemas: testSchemas(),
+			},
+			nestedStateOutput,
+		},
+		{
+			&format.StateOpts{
+				State:   deposedState(t),
+				Color:   color,
+				Schemas: testSchemas(),
+			},
+			deposedNestedStateOutput,
+		},
+		{
+			&format.StateOpts{
+				State:   onlyDeposedState(t),
+				Color:   color,
+				Schemas: testSchemas(),
+			},
+			onlyDeposedOutput,
+		},
+		{
+			&format.StateOpts{
+				State:   stateWithMoreOutputs(t),
+				Color:   color,
+				Schemas: testSchemas(),
+			},
+			stateWithMoreOutputsOutput,
+		},
```

Comment:
If we care about the order in which we traverse the module graph, it'd be nice to have a test case for that. Something like the following modules with a resource or two in each:

```mermaid
graph TD
    root --> alpha
    root --> beta
    alpha --> gamma
    alpha --> delta
    beta --> epsilon
```

(Only bringing this up because of my confusion over the comment, noted above.)

Reply:
We don't care about the order (or at least the old renderer didn't care about the order). I've replied to your comment above as well, but the tl;dr is that comment was left over.

Given we don't care about the ordering, I don't think I'll add this test. But if you think we should start caring about the order then I can add a test. Let me know!

Reply:
Nope! Let's not make more work for ourselves.